manta-js/package/sdk.ts

```diff
@@ -0,0 +1,626 @@
+
+        return {
+            api
+        };
+    }
+
+    /// Private helper method for internal use to initialize the initialize manta-wasm-wallet.
+    static async _initWasmSdk(api: ApiPromise): Promise<InitWasmResult> {
+        const wasm = await import('manta-wasm-wallet');
+        const wasmSigner = new wasm.Signer(SIGNER_URL);
+        const wasmApiConfig = new ApiConfig(
+            api, DEFAULT_PULL_SIZE, DEFAULT_PULL_SIZE
+        );
+        const wasmApi = new Api(wasmApiConfig);
+        const wasmLedger = new wasm.PolkadotJsLedger(wasmApi);
+        const wasmWallet = new wasm.Wallet(wasmLedger, wasmSigner);
+        return {
+            wasm,
+            wasmWallet,
+            wasmApi
+        }
+    }
+
+    async _setPolkadotSigner(polkadotSigner: Signer, polkadotAddress:Address): Promise<void> {
+        try {
+
+            this.wasmApi.setExternalAccountSigner(polkadotAddress);
+            this.api.setSigner(polkadotSigner);
+
+        } catch (e) {
+            console.error("Unable to set polkadotJS signer.", e);
+        }
+    }
+
+    ///
+    /// SDK methods
+    ///
+
+    /// Convert a private address to JSON.
+    convertPrivateAddressToJson(address: string): any {
+        return privateAddressToJson(address);
+    }
+
+    /// Returns information about the currently supported networks.
+    networks(): any {
+        return config.NETWORKS;
+    }
+
+    /// Returns whether the current wasm wallet instance is busy. To avoid multiple
+    /// calls to wasmWallet simultaneously which causes a borrowMut rust error.
+    isWalletBusy(): boolean {
+        return this.walletIsBusy;
+    }
+
+    ///
+    /// Manta Signer methods
+    ///
+
+    /// Returns the zkAddress of the currently connected manta-signer instance.
+    async privateAddress(): Promise<Address> {
+        if (!this.walletIsBusy) {
+            this.walletIsBusy = true;
+            const privateAddress = await getPrivateAddress(this.wasm, this.wasmWallet, this.network);
+            this.walletIsBusy = false;
+            return privateAddress
+        } else {
+            console.error("Wallet is currently busy!");
+        }
+    }
+
+    /// Performs full wallet recovery. Restarts `self` with an empty state and
+    /// performs a synchronization against the signer and ledger to catch up to
+    /// the current checkpoint and balance state.
+    ///
+    /// Requirements: Must be called once after creating an instance of MantaPrivateWallet 
+    /// and must be called before walletSync(). Must also be called after every 
+    /// time the network is changed.
+    async initalWalletSync(): Promise<void> {
+        if (!this.walletIsBusy) {
+            this.walletIsBusy = true;
+            await initSync(this.wasm, this.wasmWallet, this.network);
+            this.walletIsBusy = false;
+        } else {
+            console.error("Wallet is currently busy!");
+        }
+    }
```

Comment:
Also, we need to put everything that accesses the wasm / sets `this.walletIsBusy = true` in a try-catch block. This stuff can fail if  the signer becomes disconnected. In the `catch` block, we will want to set `walletIsBusy ` to false, so that we aren't stuck forever if we error out once.

Again, see:
 https://github.com/Manta-Network/manta-front-end/blob/main/src/contexts/privateWalletContext.tsx